ipq806x: adds kernel 4.19 support

Ansuel opened the pull request:
99% of the work to port the patch to 4.19 is from @chunkeey 
I'm proposing this as from the testing of me and some other 4.19 doesn't look to cause any regression but instead fix cpuidle. 

We have long tested this with a r7800 (me) and r7500
Think we should at least push a initial patchset for testing so that we can test this with more userbase. 

Some changes are done to usb.mk to reflect use of new qcom dwc3 driver instead of the generic one. 
This is the only way i found to use both the driver in the ipq806x Makefile.

Based on https://git.openwrt.org/?p=openwrt/staging/chunkeey.git;a=commit;h=c096b58311b52068cfa90d437d1bcfaf7fcd0d2e

Fixed broken tsens driver
Fixed bootloop on ipq8065 soc with CPU_IDLE enabled

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

pkgadd commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -0,0 +1,530 @@
+		led-failsafe = &power_amber;
+		led-running = &power_white;
+		led-upgrade = &power_amber;
+	};
```
f8d96485298e15879b34eb49a00af020aa91d334 has added a label MAC definition for kernel 4.14:
`		label-mac-device = &gmac2;`

pkgadd commented on target/linux/ipq806x/config-4.19:
```diff
@@ -0,0 +1,518 @@
+CONFIG_CRYPTO_ACOMP2=y
+CONFIG_CRYPTO_AEAD=y
+CONFIG_CRYPTO_AEAD2=y
+CONFIG_CRYPTO_CBC=y
```
9de9e9c6c504182f110c89ea472ea1eea1e69488 has removed the unsupported hw-crypto qce driver symbols for kernel 4.14, I think this needs to be replicated for config-4.19.
`
for i in CBC CTR DES DEV_QCE ECB GF128MUL NULL SEQIV XTS; do sed -i "/^CONFIG_CRYPTO_${i}=y/d" target/linux/ipq806x/config-4.19 ; done`

pkgadd commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-c2600.dts:
```diff
@@ -0,0 +1,478 @@
+		led-failsafe = &general;
+		led-running = &power;
+		led-upgrade = &general;
+	};
```
f8d96485298e15879b34eb49a00af020aa91d334 has added a label MAC definition for kernel 4.14:
`		label-mac-device = &gmac2;`

Ansuel commented:
@pkgadd if you have other comment just tell me

facboy commented:
does this just copy over the clk patchset from 4.14 (0049-0056)?  because it doesn't work that well there, and it appears to not work properly here either :(.

pkgadd commented:
I've been running this revised pull request on my ZyXEL NBG6817/ IPQ8065 for the last two days (kernel 4.19.78) - it seems to be fine (including USB, but I haven't checked this in detail). Thanks a lot for pushing this forward.

Ansuel commented:
@facboy saddly yes... as long as it doesn't cause problem we need to keep it this way waiting you for a port :D

@pkgadd we need more test...

facboy commented:
actually bandwidth is lower on this kernel, noticeably so on the `mbw` test.  in the DUMB test it can drop down to 35Mb/s, down from 170-180Mb/s.  the other 2 tests are about the same as master 4.14, so no real concern there.  i am going to see if porting the patches from linux-next plus the one left over one makes it all magically work.

Ansuel commented:
i'm here to test what you want @facboy

facboy commented:
after further wandering around `linux-next`, it looks way too complicated to me and the last patch that actually adds support for krait is AWOL and will not apply cleanly to `linux-next` either.  i'm also not convinced it will do anything for the L2 cache anyway.  i think i will try to apply my patch on top of this (leaving the existing patches this time) and see where that gets us.

btw I think those old operating-points-v2 (which you've removed) were for ipq4019 (at 716Mhz) and definitely wrong for ipq806x.

Ansuel commented:
some guy on the forum was working on them so i think you should ask to him

Gram1k commented:
Just applied this PR on R7800 and my unscientific testing with Speedtest® shows consistent 100Mbps increase in both DL/UL from 750Mbps/750Mbps to 850Mbps/850Mbps. Got 1Gbps DL/UL line from ISP.

Appreciate the speed increase. Otherwise looks ok so far.

Ansuel commented:
Any news on this ? On forum we have plenty of positive feedback from 4 different type of router

Gram1k commented:
For my pretty standard use case it's been working very well ever since installation and the logs are clean.

root@R7800:/# uname -a && uptime
Linux R7800 4.19.78 #0 SMP Fri Oct 11 08:37:08 2019 armv7l GNU/Linux
 12:29:49 up 13 days, 14:23,  load average: 0.01, 0.07, 0.07

huaracheguarache commented:
I have been running this with the latest master commits applied on top for the past day or so without any issues. USB works as well, but I have not really tested in detail. @Ansuel here is the bootlog if you want to take a look: https://gist.github.com/huaracheguarache/d4e90df68fb00f95ff99b645d4486609

ddimension commented:
I also did several tests on the 4.19 ipq806x (NBG6817). I noticed that VPN throuput (IPsec Performance) dropped to 30% with exactly the same config (kernel and openwrt), kernel crypto and arm crypto stuff tested (enabled in kconfig).
Also the whole system feels a bit slowier (execution time of binaries). Timers (clock, ddr) in dts have been compared, also checked in the running system. I'm a fan of 419, but this needs be solved before and I have no clue where to look. Any ideas?

Ansuel commented:
I think the slow down I related to cpu and cache problems and not on the
kernel

Il Ven 1 Nov 2019, 10:24 ddimension <notifications@github.com> ha scritto:

> I also did several tests on the 4.19 ipq806x (NBG6817). I noticed that VPN
> throuput (IPsec Performance) dropped to 30% with exactly the same config
> (kernel and openwrt), kernel crypto and arm crypto stuff tested (enabled in
> kconfig).
> Also the whole system feels a bit slowier (execution time of binaries).
> Timers (clock, ddr) in dts have been compared, also checked in the running
> system. I'm a fan of 419, but this needs be solved before and I have no
> clue where to look. Any ideas?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/2472?email_source=notifications&email_token=AE2ZMQU2I22G6SZ3AF3VU4TQRPYTLA5CNFSM4I56ADB2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEC2NHHQ#issuecomment-548721566>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQUWHYC6CJFHX66BDG3QRPYTLANCNFSM4I56ADBQ>
> .
>

ddimension commented:
Could you give me a hint where to start searching? I have access to multiple NBG6817.
Nevertheless thank you for this patch!


> I think the slow down I related to cpu and cache problems and not on the kernel Il Ven 1 Nov 2019, 10:24 ddimension <notifications@github.com> ha scritto:
> […](#)
> I also did several tests on the 4.19 ipq806x (NBG6817). I noticed that VPN throuput (IPsec Performance) dropped to 30% with exactly the same config (kernel and openwrt), kernel crypto and arm crypto stuff tested (enabled in kconfig). Also the whole system feels a bit slowier (execution time of binaries). Timers (clock, ddr) in dts have been compared, also checked in the running system. I'm a fan of 419, but this needs be solved before and I have no clue where to look. Any ideas? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#2472?email_source=notifications&email_token=AE2ZMQU2I22G6SZ3AF3VU4TQRPYTLA5CNFSM4I56ADB2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEC2NHHQ#issuecomment-548721566>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AE2ZMQUWHYC6CJFHX66BDG3QRPYTLANCNFSM4I56ADBQ> .

Ansuel commented:
@facboy s trying to port cache fix to kernel 4.19

wulfy23 commented:
[  219.417917] ======================================================
[  219.418447] WARNING: possible circular locking dependency detected
[  219.424443] 4.19.76 #0 Not tainted
[  219.430597] ------------------------------------------------------
[  219.433988] sh/8309 is trying to acquire lock:
[  219.440152] 0a53be28 (pernet_ops_rwsem){+.+.}, at: register_netdevice_notifier+0x28/0x1e0

[  220.049966]  Possible unsafe locking scenario:
[  220.049966] 
[  220.061764]        CPU0                    CPU1
[  220.067663]        ----                    ----
[  220.072176]   lock(&led_cdev->trigger_lock);
[  220.076692]                                lock(triggers_list_lock);
[  220.081207]                                lock(&led_cdev->trigger_lock);
[  220.087545]   lock(pernet_ops_rwsem);
[  220.094224] 
[  220.094224]  *** DEADLOCK ***

Ansuel commented:
@wulfy23 are you sure this is related to this kernel ?

wulfy23 commented:
Yes. Although it is likely benign...( may exist also in 4.14 ) and not directly related to your patchset.

It's triggered once on boot/driver init, with possibly no i'll effects.

Only posted it for brevvity.... Like a belly button, better in than out... ( no offense anyone )

EXSERENS commented:
```
[35751.683501] ------------[ cut here ]------------
[35751.683657] WARNING: CPU: 0 PID: 0 at target-arm_cortex-a15+neon-vfpv4_musl_eabi/linux-ipq806x_generic/ath10k-ct-2019-11-07-080cd781/ath10k-5.4/txrx.c:120 ath10k_txrx_tx_unref+0x494/0x68c [ath10k_core]
[35751.687324] Invalid VHT rate, nss: 3  hw_rate: 15 ratecode: 255
[35751.705079] Modules linked in: pppoe ppp_async iptable_nat ipt_MASQUERADE ath10k_pci ath10k_core ath xt_state xt_nat xt_conntrack xt_REDIRECT xt_FLOWOFFLOAD xt_CT rndis_host qmi_wwan pppox ppp_generic nf_nat_ipv4 nf_nat nf_flow_table_hw nf_flow_table nf_conntrack_rtcache nf_conntrack mac80211 ipt_REJECT cfg80211 cdc_ether xt_time xt_tcpudp xt_socket xt_multiport xt_mark xt_mac xt_limit xt_comment xt_TPROXY xt_TCPMSS xt_LOG wireguard usbnet slhc nf_tproxy_ipv6 nf_tproxy_ipv4 nf_socket_ipv6 nf_socket_ipv4 nf_reject_ipv4 nf_log_ipv4 nf_defrag_ipv6 nf_defrag_ipv4 iptable_mangle iptable_filter ip_tables crc_ccitt compat cdc_wdm tcp_bbr xt_set ip_set_list_set ip_set_hash_netportnet ip_set_hash_netport ip_set_hash_netnet ip_set_hash_netiface ip_set_hash_net ip_set_hash_mac ip_set_hash_ipportnet ip_set_hash_ipportip
[35751.760170]  ip_set_hash_ipport ip_set_hash_ipmark ip_set_hash_ip ip_set_bitmap_port ip_set_bitmap_ipmac ip_set_bitmap_ip ip_set nfnetlink nf_log_ipv6 nf_log_common ip6table_mangle ip6table_filter ip6_tables ip6t_REJECT x_tables nf_reject_ipv6 msdos ip6_udp_tunnel udp_tunnel vfat fat cifs nls_utf8 nls_iso8859_1 nls_cp437 sha512_generic seqiv md5 md4 ecb des_generic ctr cmac ccm arc4 leds_gpio dwc3 dwc3_qcom ahci sd_mod ahci_platform libahci_platform libahci libata scsi_mod gpio_button_hotplug f2fs ext4 mbcache jbd2 exfat crc32c_generic crc32_generic
[35751.808743] CPU: 0 PID: 0 Comm: swapper/0 Not tainted 4.19.82 #0
[35751.830855] Hardware name: Generic DT based system
[35751.837129] [<c030f0f4>] (unwind_backtrace) from [<c030b590>] (show_stack+0x14/0x20)
[35751.841727] [<c030b590>] (show_stack) from [<c082d660>] (dump_stack+0x84/0x98)
[35751.849630] [<c082d660>] (dump_stack) from [<c031d860>] (__warn+0xd4/0xf0)
[35751.856651] [<c031d860>] (__warn) from [<c031d4f4>] (warn_slowpath_fmt+0x34/0x4c)
[35751.863576] [<c031d4f4>] (warn_slowpath_fmt) from [<bf6c35d0>] (ath10k_txrx_tx_unref+0x494/0x68c [ath10k_core])
[35751.871247] [<bf6c35d0>] (ath10k_txrx_tx_unref [ath10k_core]) from [<bf6bca34>] (ath10k_htt_t2h_msg_handler+0x670/0x10f8 [ath10k_core])
[35751.881039] [<bf6bca34>] (ath10k_htt_t2h_msg_handler [ath10k_core]) from [<bf7182a0>] (ath10k_pci_htt_rx_cb+0x1d0/0x230 [ath10k_pci])
[35751.893212] [<bf7182a0>] (ath10k_pci_htt_rx_cb [ath10k_pci]) from [<bf6dde20>] (ath10k_ce_per_engine_service+0xa0/0x114 [ath10k_core])
[35751.905388] [<bf6dde20>] (ath10k_ce_per_engine_service [ath10k_core]) from [<bf6ddf0c>] (ath10k_ce_per_engine_service_any+0x78/0xb4 [ath10k_core])
[35751.917320] [<bf6ddf0c>] (ath10k_ce_per_engine_service_any [ath10k_core]) from [<bf71a93c>] (ath10k_pci_napi_poll+0x80/0x134 [ath10k_pci])
[35751.930376] [<bf71a93c>] (ath10k_pci_napi_poll [ath10k_pci]) from [<c06ff85c>] (net_rx_action+0x144/0x31c)
[35751.942769] [<c06ff85c>] (net_rx_action) from [<c03021b0>] (__do_softirq+0xc8/0x248)
[35751.952400] [<c03021b0>] (__do_softirq) from [<c0321ab8>] (irq_exit+0xa0/0x108)
[35751.960304] [<c0321ab8>] (irq_exit) from [<c03642b8>] (__handle_domain_irq+0xa8/0xc8)
[35751.967331] [<c03642b8>] (__handle_domain_irq) from [<c056f0f8>] (gic_handle_irq+0x6c/0xb8)
[35751.975314] [<c056f0f8>] (gic_handle_irq) from [<c0301a0c>] (__irq_svc+0x6c/0x90)
[35751.983464] Exception stack(0xc0b01f08 to 0xc0b01f50)
[35751.991121] 1f00:                   00000000 00002084 1cf55000 dd992940 c0b00000 c0b04c6c
[35751.996165] 1f20: 00002084 00002084 00000000 dd7daa00 162df720 166464e0 f9d33120 c0b01f58
[35752.004311] 1f40: c06a5e24 c06a5c98 20000013 ffffffff
[35752.012476] [<c0301a0c>] (__irq_svc) from [<c06a5c98>] (cpuidle_enter_state+0x17c/0x334)
[35752.017516] [<c06a5c98>] (cpuidle_enter_state) from [<c0347564>] (do_idle+0x190/0x220)
[35752.025664] [<c0347564>] (do_idle) from [<c0347868>] (cpu_startup_entry+0x1c/0x20)
[35752.033391] [<c0347868>] (cpu_startup_entry) from [<c0a00d90>] (start_kernel+0x480/0x490)
[35752.041076] ---[ end trace aef359e8427a8b9c ]---
```
just provide some information, maybe its just ath10k-ct problem.

ynezz commented:
@chunkeey What about this? :-) I can help merging it in the current state if you don't have time for this currently and it can be polished later (if needed at all), right?

ynezz commented:
What is needed from my point of view in order to merge it if @chunkeey is not planning to do it himself:

* split that one big commit into few smaller ones
  * one for copy of the files from the 4.14 to 4.19
  * second for updating the kernel config
  * third for refreshing the patches
  * any other commit(s) for added file(s) (if needed at all)
* proper authorship/credits
  * add @chunkeey `From:` and `Signed-off-by:` to every commit
  * keep your Signed-off-by: as well

ynezz commented on target/linux/ipq806x/Makefile:
```diff
@@ -12,6 +12,7 @@ MAINTAINER:=John Crispin <john@phrozen.org>
 SUBTARGETS:=generic
 
 KERNEL_PATCHVER:=4.14
+KERNEL_TESTING_PATCHVER:=4.19
```
I would prefer switching to 4.19 directly once it gets merged.

Ansuel commented:
Ok then will split the commit...

Il Gio 28 Nov 2019, 15:42 Petr Štetiar <notifications@github.com> ha
scritto:

> *@ynezz* commented on this pull request.
> ------------------------------
>
> In target/linux/ipq806x/Makefile
> <https://github.com/openwrt/openwrt/pull/2472#discussion_r351815690>:
>
> > @@ -12,6 +12,7 @@ MAINTAINER:=John Crispin <john@phrozen.org>
>  SUBTARGETS:=generic
>
>  KERNEL_PATCHVER:=4.14
> +KERNEL_TESTING_PATCHVER:=4.19
>
> I would prefer switching to 4.19 directly once it gets merged.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/2472?email_source=notifications&email_token=AE2ZMQS7473GRDPKXQ6FOJDQV7KD7A5CNFSM4I56ADB2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCNKN7QA#pullrequestreview-324329408>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQUIGLJIDOHKDBZOLG3QV7KD7ANCNFSM4I56ADBQ>
> .
>

Ansuel committed:
ipq806x: copy files from 4.14 to 4.19

This copies files from files-4.14 to files-4.19 directory in order to
get clear diffs for any changes done from now on.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[added missing commit description, refreshed ea8500 and wpq864 DTS]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel committed:
kernel: generic: add missing config symbol to 4.19

PCI_V3_SEMI config symbol was found missing in generic kernel config
after ipq806x config refresh to 4.19.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[commit subject and description facelift]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel commented:
@ynezz tell me if now the commits are properly split and with the right signed-off 

I don't know where to put the From: tag

adrianschmutzler commented:
> I don't know where to put the From: tag

git commit --amend --author "Name Surname <mailaddress>"

adrianschmutzler commented:
@Ansuel You might need to add a proper commit description for some of the commits.

Particularly for the "update dts files" one, I'd be interested in whether you just copied 4.14 or changed something ...

Ansuel commented:
> > I don't know where to put the From: tag
> 
> `git commit --amend --author "Name Surname <mailaddress>"`

Does this replace the author?

adrianschmutzler commented:
@Ansuel It replaces the author, which is the "From:" line when viewed as patch.

chunkeey committed:
ipq806x: refresh 4.19 patches

Reworked:

 - 0034 patchset update

Added:

 - 080 Add support for pinctrl-msm framework

Removed:

 - 0074-ipq806x-usb-Control-USB-master-reset.patch
   (we now have a dedicated driver for qcom usb)

 - 0047-mtd-nand-Create-a-BBT-flag-to-access-bad-block-marke
   (merged upstream)

 - 310-msm-adhoc-bus-support
   (it looks like it was never actually used in any dts)

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[commit subject and description facelift, SoB fix]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel committed:
ipq806x: fix tsens driver in 4.19

Added patch:

 - 063-3 Fix tsense shared memory problem

Recent changes in ioremap_resource function are causing fails for the
memory areas which are already mapped. This changes are causing tsense
driver failures during initialization:

  qcom-tsens 900000.thermal-sensor: tsens init failed

So this patch uses simple ioremap in order to use this shared memory
space.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[commit subject and desciption facelitf]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel commented:
@adrianschmutzler should be good now

adrianschmutzler commented:
The Signed-off-by should be in chronologic order from top to bottom. (So, "From:" is always the same as the first Signed-off-by). So, at least for the first commit you need to switch the Signed-off (I haven't checked the others).

hnyman commented:
@Ansuel 
Is it intentional that end result of your new split commits differ from the previous single commit in the PR?

To me it looks like there is now QCA8k stuff included, some USB related changes and also at least one wrongly formatted kernel config line (without # in the beginning):
`CONFIG_NET_DSA_VITESSE_VSC73XX is not set`

This is the diff compared to the PR two days ago:
```
diff --git a/package/kernel/linux/modules/usb.mk b/package/kernel/linux/modules/usb.mk
index d79d3c9d28..2a5f1d7d5f 100644
--- a/package/kernel/linux/modules/usb.mk
+++ b/package/kernel/linux/modules/usb.mk
@@ -72,7 +72,7 @@ $(eval $(call KernelPackage,usb-phy-nop))
 
 define KernelPackage/usb-phy-qcom-dwc3
   TITLE:=DWC3 USB QCOM PHY driver
-  DEPENDS:=@!LINUX_4_19 @(TARGET_ipq40xx||TARGET_ipq806x) +kmod-usb-dwc3-of-simple
+  DEPENDS:=@(TARGET_ipq40xx||TARGET_ipq806x) +kmod-usb-dwc3-of-simple
   KCONFIG:= CONFIG_PHY_QCOM_DWC3
   FILES:= \
     $(LINUX_DIR)/drivers/phy/phy-qcom-dwc3.ko@lt4.13 \
diff --git a/target/linux/ipq806x/Makefile b/target/linux/ipq806x/Makefile
index 40a0126ae0..571d6c2e17 100644
--- a/target/linux/ipq806x/Makefile
+++ b/target/linux/ipq806x/Makefile
@@ -22,7 +21,6 @@ DEFAULT_PACKAGES += \
        kmod-ata-core kmod-ata-ahci kmod-ata-ahci-platform \
        kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport \
        kmod-usb3 kmod-usb-dwc3-qcom \
-       kmod-usb-dwc3-of-simple kmod-usb-phy-qcom-dwc3 \
        kmod-ath10k-ct wpad-basic \
        uboot-envtools
 
diff --git a/target/linux/ipq806x/config-4.19 b/target/linux/ipq806x/config-4.19
index 06e6957a09..7fe97ac3b2 100644
--- a/target/linux/ipq806x/config-4.19
+++ b/target/linux/ipq806x/config-4.19
@@ -322,11 +322,10 @@ CONFIG_MUTEX_SPIN_ON_OWNER=y
 CONFIG_NEED_DMA_MAP_STATE=y
 CONFIG_NEON=y
 CONFIG_NET_DSA=y
-# CONFIG_NET_DSA_LEGACY is not set
-# CONFIG_NET_DSA_QCA8K is not set
-# CONFIG_NET_DSA_REALTEK_SMI is not set
-# CONFIG_NET_DSA_TAG_QCA is not set
-# CONFIG_NET_DSA_VITESSE_VSC73XX is not set
+CONFIG_NET_DSA_HWMON=y
+CONFIG_NET_DSA_QCA8K=y
+CONFIG_NET_DSA_TAG_QCA=y
+CONFIG_NET_DSA_VITESSE_VSC73XX is not set
 CONFIG_NET_FLOW_LIMIT=y
 CONFIG_NET_PTP_CLASSIFY=y
 CONFIG_NET_SWITCHDEV=y
```

Ansuel commented:
@hnyman except for the `CONFIG_NET_DSA_VITESSE_VSC73XX is not set`
all the other changes are intentional 

The dsa config are enabled in 4.14  (but never used in the dts so they are not actually included in the kernel, correct me if i'm wrong)
The changes in the usb.mk are no longer needed as we jump directly from 4.14 to 4.19, they were included to permit the use of both kernel 4.14 and 4.19 by including both the kmod need by the 2 kernel version. So I just dropped the "compatibility" changes.

adrianschmutzler commented:
As long as we do not remove 4.14 entirely, someone might still try to build with it. So, I'd the compatibility changes for now and mark them properly, so they can be removed when 4.14 is removed.

Ansuel committed:
kernel: make dwc3 usb driver depends on kernel 4.14

- usb-dwc3-of-simple is not used anymore as we have qcom dedicated driver
- usb-phy-qcom-dwc3 is not dependent of dwc3-of-simple

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@adrianschmutzler i made the 2 kernel module dependent of kernel 4.14 in case one day we will switch to kernel 5.4 
Should i add comments also in the usb.mk file?

adrianschmutzler commented:
No comment needed in usb.mk, the dependency should be self-explaining and greppable.

BTW: Why do we include kmod-usb-dwc3-of-simple in DEFAULT_PACKAGES when it is selected by kmod-usb-phy-qcom-dwc3 anyway?

Ansuel commented:
@adrianschmutzler i don't really know... i just think that usb-phy-qcom-dwc3 is implemented in a bad way... I mean AFAIK phy-qcom-dwc3 is not dependent of dwc3-of-simple... Will do some test to make sure they can insmod separately

adrianschmutzler commented:
@Ansuel Save your time. We won't use it anymore on 4.19, so no need to test just for including/not including a package that we will removed soon anyway.

Ansuel commented:
I'm actually testing if we can keep the phy-qcom-dwc3 as a module...
I your option should we include it in the kernel or not?

adrianschmutzler commented:
> I'm actually testing if we can keep the phy-qcom-dwc3 as a module...
> In your option should we include it in the kernel or not?

I cannot judge that, someone else should comment on that.

Ansuel commented:
Ok with some testing...
With phy-qcom-dwc3 as module usb ports works but in the logs there are some errors... So better include it the kernel.

hnyman commented:
Thanks,  also the new split patch series seems to work ok (with R7800 and version r11607-c7fb12beb1)

Ansuel commented:
@hnyman can you test usb ?

hnyman commented:
Both USB ports on R7800 seem to work ok with a memory stick.
```
[ 1256.523579] usb 3-1: new high-speed USB device number 2 using xhci-hcd
[ 1256.738957] usb-storage 3-1:1.0: USB Mass Storage device detected
[ 1256.740166] scsi host1: usb-storage 3-1:1.0
[ 1257.755887] scsi 1:0:0:0: Direct-Access     -Pretec  01GB             2.00 PQ: 0 ANSI: 2
[ 1257.758344] sd 1:0:0:0: [sda] 2015231 512-byte logical blocks: (1.03 GB/984 MiB)
[ 1257.763600] sd 1:0:0:0: [sda] Write Protect is off
[ 1257.770443] sd 1:0:0:0: [sda] Mode Sense: 00 00 00 00
[ 1257.775777] sd 1:0:0:0: [sda] Asking for cache data failed
[ 1257.780146] sd 1:0:0:0: [sda] Assuming drive cache: write through
[ 1257.920948]  sda: sda1
[ 1257.924284] sd 1:0:0:0: [sda] Attached SCSI removable disk
[ 1321.364556] usb 3-1: USB disconnect, device number 2
[ 1321.463697] FAT-fs (sda1): unable to read boot sector to mark fs as dirty

[ 1360.633991] usb 1-1: new high-speed USB device number 2 using xhci-hcd
[ 1360.848297] usb-storage 1-1:1.0: USB Mass Storage device detected
[ 1360.850043] scsi host1: usb-storage 1-1:1.0
[ 1361.914424] scsi 1:0:0:0: Direct-Access     Pretec   UltimateGuard    2.00 PQ: 0 ANSI: 2
[ 1361.917097] sd 1:0:0:0: [sda] 7897088 512-byte logical blocks: (4.04 GB/3.77 GiB)
[ 1361.922138] sd 1:0:0:0: [sda] Write Protect is off
[ 1361.929129] sd 1:0:0:0: [sda] Mode Sense: 00 00 00 00
[ 1361.934355] sd 1:0:0:0: [sda] Asking for cache data failed
[ 1361.938772] sd 1:0:0:0: [sda] Assuming drive cache: write through
[ 1362.114935]  sda: sda1
[ 1362.284933] sd 1:0:0:0: [sda] Attached SCSI removable disk
[ 1362.509479] FAT-fs (sda1): Volume was not properly unmounted. Some data may be corrupt. Please run fsck.

```

fantom-x commented:
Should this be added to the PR: https://github.com/openwrt/openwrt/pull/632/commits/00bee4757312da83f040e4eadac1e82e01d3af65

I have not found it here.

Ansuel commented:
@hnyman i added a test commit to this can you test if you have an improvement of ANY type 
(in theory it should fix a problem with memory performance with high cpu rates)

hnyman commented:
I can try testing it, but just one question related to this L2 issue:
This L2 cache scaling issue is a quite separate thing from the actual kernel 4.19 bump, right?

Just asking because I would not want to see the kernel bump to get bogged down by trying to solve all possible ipq806x performance issues at the same time. That could prolong things and delay the merge of the kernel bump.  So, I hope to see at least the actual kernel bump, and then the further performance improvements later.

Ansuel commented:
@hnyman you are right on everything... It's just a minimal change but better have a clean patch... 
I'm posting this to remove any regression from 4.14 as @facboy said there is a 30% worse performance in memory with 4.19

Ansuel commented:
Any news on this?

ynezz reviewed (commented):
All commits should've commit description.

ynezz commented:
> @hnyman you are right on everything... It's just a minimal change but better have a clean patch...
> I'm posting this to remove any regression from 4.14 as @facboy said there is a 30% worse performance in memory with 4.19

Lets start from the beginning, just do a 4.19 bump, wait for regressions (if any) start adding more on the top. Otherwise it would be harder to find out working/non-working version.

ynezz commented:
```
Added
- 063-3 Fix tsense shared memory problem
```
I can see that, but I don't know answer to `Why?` you're adding that. Please be more verbose.

ynezz commented on target/linux/generic/config-4.19:
```diff
@@ -3681,6 +3681,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_PCIE_KIRIN is not set
 # CONFIG_PCIE_PTM is not set
 # CONFIG_PCIE_XILINX is not set
+# CONFIG_PCI_V3_SEMI is not set
```
`kernel: generic: add missing configuration option`

Adding this config option, because foo...

ynezz commented on target/linux/ipq806x/patches-4.19/0055-cpufreq-dt-Add-L2-frequency-scaling-support.patch:
```diff
@@ -43,6 +43,13 @@ Signed-off-by: Georgi Djakov <georgi.djakov@linaro.org>
 +			l2_freq = clk_get_rate(l2_clk);
 +
 +			if (l2_freq != new_l2_freq) {
++				/* set l2_freq to idle before setting target l2_freq - there seems 
```
This belongs to commit description, but please remove this patch if its not needed to get 4.19 working.

ynezz commented on target/linux/ipq806x/Makefile:
```diff
@@ -11,16 +11,21 @@ CPU_SUBTYPE:=neon-vfpv4
 
 KERNELNAME:=zImage Image dtbs
 
+# kmod-usb-dwc3-of-simple kmod-usb-phy-qcom-dwc3 are compiled only 
```
This belongs to the commit description.

ynezz commented on target/linux/ipq806x/patches-4.19/0031-mtd-add-SMEM-parser-for-QCOM-platforms.patch:
```diff
@@ -18,7 +18,7 @@ Signed-off-by: Ram Chandra Jangir <rjangi@codeaurora.org>
 
```
I can see what you've added and removed, but important is to know `Why?` you've added it and removed it. And you're missing your `Signed-off-by:` which should be the last one, after Christian's.

ynezz commented on target/linux/ipq806x/config-4.19:
```diff
@@ -1,12 +1,19 @@
 CONFIG_ALIGNMENT_TRAP=y
```
Missing commit description.

Ansuel replied:
About kernel config i don't really tell why 

Should i try to use the one from 4.14 ?

ynezz replied:
Subject:  `ipq806x: update kernel config for 4.19`

> Why are these changes needed?

These changes are needed in order to migrate old kernel 4.14 config to the upcoming kernel version 4.19.

Ansuel replied:
Is it good now ?

ynezz commented:
FYI, build test just passed.

pkgadd commented:
I can also confirm that the current state (up to 0922c7765fe867ad221d693c4ee24ee35f9446b3), merged into this morning's OpenWrt/ master (291d79935e) builds and works.

Tested on ipq8065, ZyXEL NBG6817.

`Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>`

dissent1 committed:
ipq806x: add missing core1 voltage tolerance

Voltage tolerance is accounted per core, not per cpu, so add
missing DT entry.

Signed-off-by: Pavel Kubelun <be.dissent@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@ynezz changes requested done

Ansuel commented:
@ddimension performance degradation could be caused by a bug in the L2 cache scaling patch... If you want i can pass you a modified patch if it does actually solve the problem.

ynezz commented:
@pkgadd @hnyman Thank you for testing! @hnyman Can I add your `Tested-by:` as well?

ynezz commented:
I've merged this into my staging tree https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=shortlog;h=refs/heads/staging

@dissent1 can you check https://github.com/ynezz/openwrt/commit/fa9b6ce56114f024365f8d5e2893a28e8d3ce2d1 ?

hnyman commented:
@ynezz 

>  @hnyman Can I add your Tested-by: as well?

Yes, you can.
I have compiled lately three versions of this PR, and all have worked ok. 

I have tested today's version only briefly, but the code in commits of r11619-f6385f30bd-20191203 is practically identical to my previous build (r11607-c7fb12beb1-20191129)

notmyrealhandle commented:
Do you want to include the patch to expand overlay for the r7500v2 (PR for 4.14 [here](https://github.com/openwrt/openwrt/pull/1922))?  

Current master, v18 and v19 all have the overlay expansion patch for the r7500v2).  Not including this patch for the r7500v2 may cause boot loops if an unaware user attempts to sysupgrade from an earlier version.

FWIW my r7500v2 is now an AP only (been running 4.19 for ~ 5 months with no issues) and I personally don't need the extra space anymore.

Ansuel commented:
@notmyrealhandle this is not already included... my bad

notmyrealhandle commented:
@ansuel,

no worries and thank you for putting in the effort to get this into master.

Ansuel commented:
@ynezz can we include this?

hnyman commented:
> the patch to expand overlay for the r7500v2

I agree with @Ansuel and @notmyrealhandle here. 
It should be included in the 4.19 patch set. Otherwise things break down for r7500v2 users.

The change has been made in master as 57b1e951b but apparently the current DTS version is not copied for 4.19 in this patch set, but an older version of that DTS file has been . Probably because chunkeey's original patch is from Feb 2019 and that change is from March.

But that made me to look at the DTS change log more closely:
https://github.com/openwrt/openwrt/commits/master/target/linux/ipq806x/files-4.14/arch/arm/boot/dts

To me it looks like at least the commit 064e431575 is also missing. 
( `ipq8064: Enabling sata port ipq8064 based devices` )

ynezz commented:
I redid the copy files-4.14 to files-4.19 step again (3826ff0b4f3c ipq806x: copy files from 4.14 to 4.19
) and got this diff:

```diff
diff --git a/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ea8500.dts b/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ea8500.dts
index 91c358d24564..ca8376afd7e3 100644
--- a/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ea8500.dts
+++ b/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ea8500.dts
@@ -39,10 +39,6 @@
 
 	soc {
 		pinmux@800000 {
-
-			pinctrl-0 = <&switch_reset>;
-			pinctrl-names = "default";
-
 			button_pins: button_pins {
 				mux {
 					pins = "gpio65", "gpio67", "gpio68";
@@ -70,16 +66,6 @@
 				};
 			};
 
-			switch_reset: switch_reset_pins {
-				mux {
-					pins = "gpio63";
-					function = "gpio";
-					drive-strength = <2>;
-					bias-disable;
-					output-low;
-				};
-			};
-
 			mdio0_pins: mdio0_pins {
 				mux {
 					pins = "gpio0", "gpio1";
@@ -178,6 +164,10 @@
 			status = "ok";
 		};
 
+		pcie2: pci@1b900000 {
+			status = "ok";
+		};
+
 		nand@1ac00000 {
 			status = "ok";
 
@@ -408,5 +398,3 @@
 	};
 };
 
-/delete-node/ &pcie2_pins;
-/delete-node/ &pcie2;
diff --git a/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts b/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts
index c246308e8c1d..14258d594e72 100644
--- a/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts
+++ b/target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts
@@ -151,7 +151,7 @@
 						reg = <0x1340000 0x4000000>;
 					};
 
-					BOOTCONFIG@1340000 {
+					BOOTCONFIG@5340000 {
 						label = "BOOTCONFIG";
 						reg = <0x5340000 0x0060000>;
 					};

```

So it looks like, that "just" following commits were missing:

* 54d9ac8ba822 ipq806x: WPQ864: fix partition unit-address
* 04d6753d03e3 Revert "ipq806x: fix EA8500 switch control"

Should be fixed now, please verify again, thanks.

ynezz commented:
> Should be fixed now, please verify again, thanks.

No, its not fixed, it's being overwrite by commit b51ab26cdd10 ("ipq806x: update dts files"). 

And while fixing the merge conflict in wpq864, I've realized, that I really don't like this big commit with "17 files changed, 801 insertions(+), 820 deletions(-)" and don't wont to merge this until its split and better reviewed.

Ansuel commented:
@ynezz how should be splitted?

ynezz commented:
> @ynezz how should be splitted?

So it makes sense and it's easier to review, as you can see some problems went in unnoticed. Those should be separate commits:

* Use new usb3 implementation
* Set qcom,apq8064-saw2-v1.1-cpu as regulator t make cpuidle work
* Disable spc on ipq8065 soc due to incompatibility

The remaining `Rework some dts (from Christian Lamparter)` should be split as well, either group them by changes (ok -> okay etc.) or by device, including proper commit description.

ynezz commented:
And ideally base it on the previous work available in https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=shortlog;h=refs/heads/staging-ipq806x-4.19 as I'm not willing to redo the remaining commits again, thanks.

chunkeey committed:
ipq806x: convert ok to okay in dts

DT spec require okay instead of ok in dts files

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: ap148: reorganize dts

Reorganize dts to use tags from ipq8064 dtsi

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: ap161: fix wrong definition stdout

From documentation
https://www.kernel.org/doc/Documentation/devicetree/bindings/chosen.txt
should be just stdout-path

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: improve mdio gpios list

Improve rediability of gpio mdio list. No functional change.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@ynezz yes there was some testing code in some dts... (Leftover code of mdio and dsa testing from chunkeey) cleared and fixed. Sorry to ask but can you re-review?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ea8500.dts:
```diff
@@ -165,7 +149,7 @@
 
 		pcie2: pci@1b900000 {
-			status = "okay";
+			status = "ok";
```
Belongs to other patch.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-r7500v2.dts:
```diff
@@ -257,7 +241,13 @@
+					netgear@3480000 {
+						label = "netgear";
+						reg = <0x3480000 0x4480000>;
+						read-only;
```
Should be removed.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -919,90 +919,84 @@
 		};
-
-		hs_phy_1: phy@100f8800 {
+		
```
whitespace issue?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -225,32 +225,16 @@
+		usb3_1: usb3@100f8800 {
 			status = "okay";
-
+			
```
whitespace issue?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -225,32 +225,16 @@
-		usb30@0 {
-			status = "okay";
-
+			
```
whitespace issue?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -558,9 +559,9 @@
-			interrupts = <0 19 0>,
-				     <0 21 0>,
-				     <0 22 0>;
+			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>,
```
You've replaced 0=IRQ_TYPE_NONE with 4=IRQ_TYPE_LEVEL_HIGH

ynezz replied:
I mean, commit description doesn't correspond with the patch content.

Ansuel replied:
About this we should really ask @chunkeey why every 0 value has been changed to 4... Restoring them... Checked some other dtsi online and they all have 0 value so... ? Error in conversion ?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -737,7 +743,7 @@
 				reg = <0x12490000 0x1000>,
 				      <0x12480000 0x1000>;
-				interrupts = <0 195 0x0>;
+				interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
```
You've replaced 0=IRQ_TYPE_NONE with 4=IRQ_TYPE_LEVEL_HIGH

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -746,7 +752,7 @@
 				compatible = "qcom,i2c-qup-v1.1.1";
 				reg = <0x124a0000 0x1000>;
-				interrupts = <0 196 0>;
+				interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>;
```
You've replaced 0=IRQ_TYPE_NONE with 4=IRQ_TYPE_LEVEL_HIGH

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -1021,7 +1027,7 @@
 				  0x82000000 0 0x08000000 0x08000000 0 0x07e00000>; /* non-prefetchable memory */
 
-			interrupts = <GIC_SPI 35 IRQ_TYPE_NONE>;
+			interrupts = <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>;
```
You've replaced IRQ_TYPE_NONE with IRQ_TYPE_LEVEL_HIGH

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065.dtsi:
```diff
@@ -75,4 +75,12 @@
 			};
 		};
 	};
+	
```
whitespace issue

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065.dtsi:
```diff
@@ -75,4 +75,12 @@
 		};
 	};
+	
+	cpus {
```
Shouldn't this part go into separate patch? I quite miss how this is related to the changes in qcom-ipq8064.dtsi

Ansuel replied:
It's related to cpuidle. Should I make 2 commit, one for cpuidle enable and one for disabling spc on ipq8065?

ynezz replied:
In order to make it clear, yes.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -633,7 +633,7 @@
 				mux {
```
Try to line wrap your commit descriptions to < 72 chars. Otherwise I would need to touch this commit again and line wrap that commit description, thanks.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -863,7 +914,7 @@
 		};
 
-		sata@29000000 {
+		sata: sata@29000000 {
```
It would make sense to add sata and nand changes into separate commit as it makes no sense in `add missing gpio and gsbi declaration` commit.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -633,7 +633,7 @@
 					pins = "gpio3";
 					function = "pcie1_rst";
-					drive-strength = <2>;
+					drive-strength = <12>;
```
2 -> 12 doesn't look like `add missing gpio and gsbi declaration`, probably separate commit with description why we need to change 2 to 12.

Ansuel replied:
Added a reason in separate commit

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064.dtsi:
```diff
@@ -695,7 +695,8 @@
 		timer@200a000 {
-			compatible = "qcom,kpss-timer", "qcom,msm-timer";
+			compatible = "qcom,kpss-timer",
+					"qcom,kpss-wdt-ipq8064", "qcom,msm-timer";
```
indent issue?

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -385,6 +415,8 @@
 				speed = <1000>;
 				full-duplex;
+				pause;
+                asym-pause;
```
indent/whitespace issue

ynezz replied:
should go to separate commit `Set asym-pause and pause support for gmac driver.`

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -405,6 +437,8 @@
 				speed = <1000>;
 				full-duplex;
+				pause;
+                asym-pause;
```
indent/whitespace issue

ynezz replied:
should go to separate commit Set asym-pause and pause support for gmac driver.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8065-r7800.dts:
```diff
@@ -241,11 +241,41 @@
 
```
line wrap commit description < 72 chars otherwise I would need to do it myself

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-ap161.dts:
```diff
@@ -25,7 +25,7 @@
 	};
```
Missing commit description.

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts:
```diff
@@ -333,6 +333,8 @@
 	fixed-link {
```
Would it be too much to know Why? are you adding those options? I can already see, that you've added them :-)

Ansuel replied:
@ynezz can't really find a reason for this change... As this was done only for some board i dropped it. (Think it was in testing by chunkeey)

ynezz commented on target/linux/ipq806x/files-4.19/arch/arm/boot/dts/qcom-ipq8064-wpq864.dts:
```diff
@@ -226,7 +226,8 @@
 
-		gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH &qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
+		gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH>,
+				<&qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
```
some indent issue?

ynezz commented on target/linux/ipq806x/patches-4.19/0063-3-tsens-fix-kernel-4_19.patch:
```diff
@@ -0,0 +1,20 @@
+ 
+ int __init init_common(struct tsens_device *tmdev)
+ {
++	resource_size_t size;
```
I just noticed this, sorry for this nitpick. I would move the definition under `struct resource *res;`, to keep christmast tree ordering.

chunkeey committed:
ipq806x: convert ipq8064 dtsi interrupts

Convert hardcoded interrupts value to types defined in gci include file.
Interrupts sets to 0 are converted to IRQ_TYPE_LEVEL_HIGH to fix
kernel warning. Same fix has been applied to arm64 dts.

https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1797143
https://patchwork.kernel.org/patch/10367453/
https://patchwork.kernel.org/patch/10315315/

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: set apq8064 regulator to support cpuidle

Set qcom,apq8064-saw2-v1.1-cpu as regulator to make cpuidle work.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: disable spc on ipq8065 soc

Spc is disabled in ipq8065 board as it does cause cpu lockup
(probably caused by wrong register being set)

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: add nand and sata tags

Add some tags for nand and sata structure to easily
reference them in other dts.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: add missing gpio and gsbi declaration

Adds missing gpio and gsbi declaration.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: increase drive-strength for pinmux

Increase drive-strength from
https://lore.kernel.org/patchwork/patch/626885/

2mA drive strength is not enough when we connect multiple i2c devices
on the bus with different pull up resistors.

This issue was detected when multiple i2c devices
connected on the other side of level shifters on Linaro sensor board.
Maxing up to 16mA made i2c much stable.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: use ipq8064 dedicated watchdog

Add missing watchdog to list of compatible timers

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel committed:
ipq806x: use new usb3 implementation

Use new usb3 implementation and refresh dts to the new dwc3 structure

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@ynezz Third time is always good right ? :D

chunkeey committed:
ipq806x: r7800: add missing wifi definition for pcie

Add missing wifi compatible definition for r7800 device.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

chunkeey committed:
ipq806x: d7800: fix warning on dtc compilation

There is warning with "property has invalid length (4 bytes)"
related to nand definition. Set size-cells to zero to fix this.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
Interesting... the change to the interrupts cause some tracelog...

```
------------[ cut here ]------------
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.045228] WARNING: CPU: 0 PID: 71 at drivers/irqchip/irq-gic.c:1016 gic_irq_domain_translate+0x6c/0x108
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.049029] Modules linked in: ahci_platform(+) libahci_platform libahci libata scsi_mod ehci_hcd gpio_button_hotplug ext4 mbcache jbd2 btrfs xor zstd_decompress zstd_compress xxhash xor_neon raid6_pq libcrc32c crc32c_generic
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.058505] CPU: 0 PID: 71 Comm: kmodloader Tainted: G        W         4.19.86 #0
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.078359] Hardware name: Generic DT based system
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.085834] [<c030f0d4>] (unwind_backtrace) from [<c030b564>] (show_stack+0x14/0x20)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.090612] [<c030b564>] (show_stack) from [<c07db444>] (dump_stack+0x90/0xa4)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.098511] [<c07db444>] (dump_stack) from [<c031d49c>] (__warn+0xd4/0xf0)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.105534] [<c031d49c>] (__warn) from [<c031d4f8>] (warn_slowpath_null+0x40/0x48)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.112392] [<c031d4f8>] (warn_slowpath_null) from [<c053a954>] (gic_irq_domain_translate+0x6c/0x108)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.119952] [<c053a954>] (gic_irq_domain_translate) from [<c03682e4>] (irq_create_fwspec_mapping+0xc4/0x344)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.129236] [<c03682e4>] (irq_create_fwspec_mapping) from [<c03685a8>] (irq_create_of_mapping+0x44/0x4c)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.139141] [<c03685a8>] (irq_create_of_mapping) from [<c069d6b0>] (of_irq_get+0x30/0x40)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.148601] [<c069d6b0>] (of_irq_get) from [<c05b9750>] (platform_get_irq+0x30/0xf8)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.156668] [<c05b9750>] (platform_get_irq) from [<bf2a19f0>] (ahci_platform_init_host+0x64/0x674 [libahci_platform])
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.164488] [<bf2a19f0>] (ahci_platform_init_host [libahci_platform]) from [<bf2a8098>] (ahci_probe+0x98/0xc0 [ahci_platform])
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.174988] [<bf2a8098>] (ahci_probe [ahci_platform]) from [<c05b9354>] (platform_drv_probe+0x40/0x84)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.186267] [<c05b9354>] (platform_drv_probe) from [<c05b7810>] (really_probe+0x15c/0x2d8)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.195553] [<c05b7810>] (really_probe) from [<c05b7ca0>] (__driver_attach+0xa8/0xe8)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.203799] [<c05b7ca0>] (__driver_attach) from [<c05b5a3c>] (bus_for_each_dev+0x50/0xa4)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.211699] [<c05b5a3c>] (bus_for_each_dev) from [<c05b6c40>] (bus_add_driver+0xf4/0x210)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.219858] [<c05b6c40>] (bus_add_driver) from [<c05b847c>] (driver_register+0xcc/0x114)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.228020] [<c05b847c>] (driver_register) from [<c0302634>] (do_one_initcall+0x8c/0x1b8)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.236183] [<c0302634>] (do_one_initcall) from [<c0390254>] (do_init_module+0x68/0x200)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.244251] [<c0390254>] (do_init_module) from [<c0392294>] (load_module+0x1e2c/0x21fc)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.252411] [<c0392294>] (load_module) from [<c03927c8>] (sys_init_module+0x164/0x198)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.260136] [<c03927c8>] (sys_init_module) from [<c0301000>] (ret_fast_syscall+0x0/0x54)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.268119] Exception stack(0xdcf09fa8 to 0xdcf09ff0)
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.276370] 9fa0:                   00000000 00000000 01ea2810 000018d4 000128fa 0000fe00
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.281322] 9fc0: 00000000 00000000 00000005 00000080 000018d4 00000000 00023d10 00000000
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.289477] 9fe0: be92cd0c be92ccf0 00011d64 b6eddf40
Thu Dec  5 02:22:55 2019 kern.warn kernel: [   16.297701] ---[ end trace e3be67b3b0aea9a2 ]---
```

hnyman commented:
The newest version works ok with r11625-a512123a4b (as of 20191206).

KanjiMonster commented on package/kernel/linux/modules/usb.mk:
```diff
@@ -72,7 +72,7 @@ $(eval $(call KernelPackage,usb-phy-nop))
 define KernelPackage/usb-phy-qcom-dwc3
   TITLE:=DWC3 USB QCOM PHY driver
-  DEPENDS:=@(TARGET_ipq40xx||TARGET_ipq806x) +kmod-usb-dwc3-of-simple
+  DEPENDS:=@(TARGET_ipq40xx||TARGET_ipq806x)
```
Don't you need to add `+LINUX_4_14:kmod-usb-dwc3-of-simple` here to not break 4.14 (which you still leave as default)?

Ansuel replied:
They are independent and dwc3-of-simple is selected anyway by the soc makefile.

ynezz replied:
Here are builds of this PR:

* https://gitlab.com/ynezz/openwrt/-/jobs/371753867 (with 4.14)
* https://gitlab.com/ynezz/openwrt/-/jobs/371876989 (with 4.19)

KanjiMonster replied:
Now please take the Image Builder (you did build an Image Builder, right?) from the 4.19 build and try to build an image for any device.

KanjiMonster commented on target/linux/ipq806x/Makefile:
```diff
@@ -20,7 +21,8 @@ DEFAULT_PACKAGES += \
 	kmod-ata-core kmod-ata-ahci kmod-ata-ahci-platform \
 	kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport \
-	kmod-usb3 kmod-usb-dwc3-of-simple kmod-usb-phy-qcom-dwc3 \
+	kmod-usb-dwc3-of-simple kmod-usb-phy-qcom-dwc3 \
```
Having `kmod-usb-dwc3-of-simple` here will  probably break imagebuilder builds with 4.19, since `kmod-usb-dwc3-of-simple' isn't available there. But if `kmod-usb-phy-qcom-dwc3` depends on it for 4.14 (as it currently does), you should be able to remove `kmod-usb-dwc3-of-simple` without any issues.

KanjiMonster replied:
```
$ make image PROFILE=netgear_r7500
Checking 'working-make'... ok.
Checking 'case-sensitive-fs'... ok.
...
Configuring urngd.
Configuring ppp-mod-pppoe.
Collected errors:
 * opkg_install_cmd: Cannot install package kmod-usb-dwc3-of-simple.
make[2]: *** [Makefile:157: package_install] Error 255
make[1]: *** [Makefile:118: _call_image] Error 2
make: *** [Makefile:197: image] Error 2
$
```

Ansuel commented:
@ynezz should we keep this as testing kernel or switch to 4.19 directly now that we have cleared everything?

KanjiMonster reviewed (changes requested):
The usb dependencies need to be resolved so ImageBuilder does not fail.

To explain, IB does not know about modules not being available, it sees `kmod-usb-dwc3-of-simple` is part of the default package set, so it tries to install that and then fails since the package does not exist.

Ansuel commented:
@KanjiMonster ok then how to solve this?

dw3-of-simple must be included only on kernel <= 4.14 
dw3-qcom must be included only on kernel > 4.14

KanjiMonster commented:
Have neither in the default packages, and use conditional dependencies for a common kmod so the appropriate per-kernel dependencies get included.

hnyman commented:
Trying to declare conditional default modules for short-term use  sounds overly complicated to me.

Could we just bypass declaring 4.19 first as a testing kernel, and directly switch to 4.19 and swap the included default usb3 package in the same Makefile commit. 

@ynezz already suggested that 9 days ago:
> I would prefer switching to 4.19 directly once it gets merged.

Then the build before that commit would have 4.14 and the old usb3, and newer builds would have 4.19 and the new usb3 module.

adrianschmutzler commented:
> Then the build before that commit would have 4.14 and the old usb3, and newer builds would have 4.19 and the new usb3 module.

We were at this point before, and then I requested introducing the condition again because kernel 4.14 is still available and someone could still try to build with that.

If you want to go that way, you would have to drop support for 4.14 entirely from ipq806x.

Maybe just try what KanjiMonster suggests.

hnyman commented:
Simplest would be to just move to 4.19 and drop the 4.14 support from ipq806x. (not introduce 4.19 as "testing-kernel" but as the main kernel version and clean-up the 4.14 config/files/patches.)

Most popular targets have already been on 4.19 for months. And the underlying kernel version bump commit for ipq806x was originally drafted by chunkeey in February , so this ipq806x bump that been in the baking for a long time.

Trying to keep 4.14 possibility alive there seems like an unnecessary complication.

pkgadd commented:
I've been experimenting with kernel 4.19 on ipq806x since early may and used it continuously since july (2019-07-19), without noticing any issues on my nbg6817/ ipq8065 at all (updating roughly every 2-3 weeks). Routing performance is significantly better with kernel 4.19, stmmac caused crashes when encountering jumbo frames on the wire are said to be gone in 4.19 (but personally I haven't seen them on 4.14 or 4.9 either). Therefore I'd suggest to keep it simple and to make a hard switch to kernel 4.19 as well, without complicating it with the testing kernel definition (kernel 4.14 patches/ files don't need to be removed immediately, giving potential users to revert by editing  `KERNEL_PATCHVER` and `DEFAULT_PACKAGES` (for the different USB modules) in `target/linux/ipq806x/Makefile`).

ynezz commented:
@424778940z @linbox1968 Can you please test this on EA8500? This should be kernel in the next release, so testing is really welcome. You can either build it yourself (don't forget to add `CONFIG_TESTING_KERNEL=y` to the .config or you'll built image with 4.14 kernel) or use prebuilt images from https://gitlab.com/ynezz/openwrt/-/jobs/371876989/artifacts/browse/bin/targets/ipq806x/generic/

shelterx commented:
My Netgear r7800 seems to run fine with these patches, running 4.19.88 (r11651-487e0631d0) now. Thanks for all the effort put into this.

ddimension commented:
I tested with Zyxel NBG6817, it works very good. I had to enable in kernel config-4.19:
CONFIG_ARM_MODULE_PLTS=y
Without that, I get vmalloc errors because of my module count. Perhaps that should be enabled, take a look: https://cateee.net/lkddb/web-lkddb/ARM_MODULE_PLTS.html

Ansuel committed:
ipq806x: update 4.19 kernel config

These changes are needed in order to migrate old kernel 4.14 config to
the upcoming kernel version 4.19.  Also add missing configuration
options that comes up with this SoC.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[line wrap long commit description, add 4.19 to subject, soc->SoC]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel committed:
ipq806x: update USB3 modules for 4.19

- Use new dwc3-qcom usb driver.
- Drop dwc3-of-simple as we have a dedicated driver now.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[split into separate commit, commit subject facelift]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel committed:
ipq806x: switch to 4.19 kernel version

It has been used by several people for some time already and feedback
has been mostly positive.

Ref: https://github.com/openwrt/openwrt/pull/2472
Tested-by: Hannu Nyman <hannu.nyman@iki.fi> [ipq8065, R7800]
Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de> [ipq8065, NBG6817]
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
[separate commit, commit subject and description facelift]
Signed-off-by: Petr Štetiar <ynezz@true.cz>

Ansuel commented:
@ynezz i updated this to switch directly to address problem with imagebuild. 
I also modified a commit that wrongly removed the compatible from device dts.
Added a config option to fix @ddimension problem. 

Any news about this?

Ansuel committed:
ipq806x: fix missing compatible and rename

- ipq8064 compatible was missing from nbg6817.
- Rename ap148 with a better descriptive name.

Signed-off-by: Christian Lamparter <chunkeey@gmail.com>
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

ynezz commented:
> Any news about this?

@Ansuel Thank you for your work and patience! :-)

Ansuel commented:
Np. I notice you readded the testing commit.
When this will get merged in master will ipq be switched to 4.19 directly?


Il Gio 19 Dic 2019, 14:13 Petr Štetiar <notifications@github.com> ha
scritto:

> Any news about this?
>
> @Ansuel <https://github.com/Ansuel> Thank you for your work and patience!
> :-)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/2472?email_source=notifications&email_token=AE2ZMQR6KKQEONQI3OZOKTLQZNXOPA5CNFSM4I56ADB2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHJR5IY#issuecomment-567484067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQUN5IOBGJ4JGDVGZE3QZNXOPANCNFSM4I56ADBQ>
> .
>
